loadfile/loadfile.go

```diff
@@ -153,18 +153,21 @@ func (fc *fileCache) Files() map[string]ContextFile {
 // The new root directory will be the root directory of the last file cache
 // argument. File keys found later in iteration will overwrite previously
 // file keys, if there is a name clash.
-func MergeFileCaches(fileCaches ...FileCache) FileCache {
+func MergeFileCaches(fileCaches ...FileCache) (FileCache, error) {
 	cache := map[string]ContextFile{}
 	rootDir := ""
 
 	for _, fc := range fileCaches {
 		for key, contextFile := range fc.Files() {
 			cache[key] = contextFile
 		}
+		if rootDir != "" && rootDir != fc.RootDir() {
+			return nil, fmt.Errorf("file caches have different root directory %q, %q", rootDir, fc.RootDir())
+		}
 		rootDir = fc.RootDir()
```

Comment:
Someday, Matt (in scrum on Tuesday, perhaps?), maybe you can explain why you're so intent to overwrite `rootDir` here with the value that it already has (as enforced by the previous `if` statement)....  🍤